packages/amplify-category-function/src/provider-utils/awscloudformation/index.ts

```diff
@@ -139,6 +141,67 @@ async function addLayerResource(
 }
 
 export async function updateResource(context, category, service, parameters, resourceToUpdate) {
+  // load the service config for this service
+  const serviceConfig: ServiceConfig<FunctionParameters> | ServiceConfig<LayerParameters> = supportedServices[service];
+  const BAD_SERVICE_ERR = `amplify-category-function is not configured to provide service type ${service}`;
+  if (!serviceConfig) {
+    throw BAD_SERVICE_ERR;
+  }
+  switch (service) {
+    case ServiceName.LambdaFunction:
+      return updateFunctionResource(context, category, service, parameters, resourceToUpdate);
+    case ServiceName.LambdaLayer:
+      return updateLayerResource(context, category, service, serviceConfig, parameters);
+    default:
+      throw BAD_SERVICE_ERR;
+  }
+}
+
+export async function updateLayerResource(
+  context,
+  category,
+  service,
+  serviceConfig: ServiceConfig<LayerParameters>,
+  parameters?: Partial<LayerParameters>,
+): Promise<void> {
+  if (!serviceConfig) {
+    throw `amplify-category-function is not configured to provide service type ${service}`;
+  }
+
+  if (!parameters) {
+    parameters = {};
+    parameters.providerContext = {
+      provider: provider,
+      service: ServiceName.LambdaLayer, // TODO: switch to using service
```

Comment:
I'd change this to `service: service` and remove the TODO comment

Reply:
done